src/Maestro/Maestro.Web/Api/v2020_02_20/Controllers/BuildsController.cs

```diff
@@ -209,6 +218,23 @@ public async Task<IActionResult> Create([FromBody, Required] BuildData build)
         {
             Data.Models.Build buildModel = build.ToDb();
             buildModel.DateProduced = DateTimeOffset.UtcNow;
+
+            if (buildModel.Incoherencies == null)
+            {
+                try
+                {
+                    buildModel.Incoherencies = await 
```

Comment:
isn't this an expensive operation to do on every every POST?

Reply:
Yeah we need to profile this to make sure that it doesn't take too long. This doesn't look up build info any longer, so it should be much, much faster than it was before, but....
- How long does it take today?
- What happens if we hit failure cases while looking up dependencies, which could cause retries, etc. and delay this?

I can see two paths:
- We move the incoherency build to the client side, prior to creating a build
- We populate incoherencies in the queue model, like we do in some cases where we expect long operations (like triggering subscriptions), and the returned build model from Create simply does not immediately include that data

Reply:
To me it doesn't sound terrible expensive. I did some profiling looking at the latest ~700 builds. Here are some info:

- Most of the builds have 0 incoherencies and the computation is fast. Less than 2 seconds.
- Some builds have lost of incoherencies and computatioin take a couple dozen of seconds.

Here is a histogram excluding the coherent builds:

![time](https://user-images.githubusercontent.com/2249648/80434780-95142080-88af-11ea-88a4-51997e2810e7.png)

Do you guys think that's too slow? TBH my opinion is that the cost of a couple of seconds for the whole build isn't much.

Reply:
I think that delaying the response to a post request by 10 seconds is a lot

Reply:
I think that for the context where the request happens, 10 seconds, or even 1 minute, isn't a lot. If we had a scenario where the user would call this API directly and wait for the response I'd fully agree that 10s is a lot. But the only scenario where this request happens is the one where responsiveness isn't a problem AFAIU.

Reply:
I'm not comfortable with that length of delay. What happens if we hit a retry and end up waiting for 2 minutes? In that case, it's highly likely we'd hit an http client timeout and then the upload would fail.

That said, this field should not need to be immediately available, as it's informational only. So I think we can simply queue this operation and return the response back to the user. The incoherencies will be filled in a bit later.

To be complete, the incoherencies field could be 3 state:
- Not available (I suppose this is the case if you uploaded using an older API version?)
- Pending calculation
- Has results.

2 state might suffice too.

Reply:
I like the idea of simply starting the operation without blocking the response.

Also. Thank you for getting the concrete data! Makes it so much easier to weigh the options.

Reply:
>Also. Thank you for getting the concrete data! Makes it so much easier to weigh the options.

Yes, thanks for getting the data.

Reply:
I'll patch the code to use a `BackgroundQueue` 'event' to fill in the Incoherencies async then. Thank you all for the feedback.